packages/hydrogen-react/src/CartProvider.tsx

```diff
@@ -288,7 +288,7 @@ export function CartProvider({
    * 2. localStorage cartId
    */
   useEffect(() => {
-    if (!cartReady.current && !fetchingFromStorage.current) {
+    if (!cartReady && !fetchingFromStorage.current) {
```

Comment:
I think part of the reason the previous implementation used a ref is because `useEffect()` executes multiple times, and `setState()` is async. The next time the effect runs the state might not have changed.